packages/front-end/components/Events/EventsPage/EventsPage.tsx

```diff
@@ -224,44 +228,38 @@ export const EventsPageContainer = () => {
         />
       </div>
       <div>
-        <Field
-          type="date"
+        <DatePicker
+          date={fromDate}
+          setDate={setFromDate}
+          scheduleEndDate={toDate}
           label="From"
-          className="text-muted"
-          labelClassName="mr-2 mb-0"
-          containerClassName="ml-2 d-flex align-items-center mb-0"
-          value={fromDate ? fromDate.toISOString().split("T")[0] : ""}
-          onChange={(d) => {
-            setFromDate(d.target.value ? new Date(d.target.value) : null);
-          }}
+          precision="date"
+          containerClassName=""
         />
       </div>
       <div>
-        <Field
-          type="date"
+        <DatePicker
+          date={toDate}
+          setDate={setToDate}
+          scheduleStartDate={fromDate}
           label="To"
-          className="text-muted"
-          labelClassName="mr-2 mb-0"
-          containerClassName="ml-2 d-flex align-items-center mb-0"
-          value={toDate ? toDate.toISOString().split("T")[0] : ""}
-          onChange={(d) => {
-            setToDate(d.target.value ? new Date(d.target.value) : null);
-          }}
+          precision="date"
+          containerClassName=""
```

Comment:
Why not use the date range functionality here instead of 2 inputs?

Reply:
The interval start/end dates are both optional, which doesn't play nicely with this widget. We could eventually add custom support for optional ranges